### PR TITLE
[Doc] Add links to external libraries: Taichi GLSL & Taichi Elements

### PR DESCRIPTION
**Pull request by archibate**
Related issue = #

[[Click here for the format server]](http://kun.csail.mit.edu:31415/)

----

### CODE REVIEW

File: docs/external_libraries.rst [hunk 1]
```diff
@@ -0,0 +1,43 @@
+External libraries
+==================
+
+The Taichi Programming Language provides a set of built-in standard library.
+The Taichi built-in library is aimed to be simple for maintainability, and
+therefore generic and less domain specific.
+
+However users may find that the built-in library along may not enough,
+so here we provide some external libraries that hopefully could extend your
+productivity in Taichi.
```
Comment (yuanming-hu) [inline docs/external_libraries.rst:10]:
```suggestion
The Taichi programming language offers a minimal and generic built-in standard library. Extra domain-specific functionalities are provided via **extension libraries**:
```

File: docs/external_libraries.rst [hunk 2]
```diff
@@ -0,0 +1,43 @@
+External libraries
+==================
+
+The Taichi Programming Language provides a set of built-in standard library.
+The Taichi built-in library is aimed to be simple for maintainability, and
+therefore generic and less domain specific.
+
+However users may find that the built-in library along may not enough,
```
Comment (rexwangcc) [inline docs/external_libraries.rst:8]:
```suggestion
However, users may find it not adequate enough for their use cases to use the built-in library alone
```

File: docs/external_libraries.rst [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+so here we provide some external libraries that hopefully could extend your
```
Comment (rexwangcc) [inline docs/external_libraries.rst:9]:
```suggestion
so here we provide some external libraries that hopefully could increase your
```

File: docs/external_libraries.rst [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+productivity in Taichi.
+
+
+Taichi GLSL
+-----------
+
+`Taichi GLSL <https://github.com/taichi-dev/taichi_glsl>`_ is an external
+library of Taichi, aimed to provide a set of useful helper functions including
+but not limited to:
```
Comment (yuanming-hu) [inline docs/external_libraries.rst:18]:
```suggestion
`Taichi GLSL <https://github.com/taichi-dev/taichi_glsl>`_ is an extension
library of Taichi, aiming at providing useful helper functions including:
```

File: docs/external_libraries.rst [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+1. Handy scalar functions like ``clamp``, ``smoothstep``, ``mix``, ``round``.
+2. GLSL-alike vector functions like ``normalize``, ``distance``, ``reflect``.
+3. Well-behaved random generators including ``randUnit2D``, ``randNDRange``.
+4. Possible Taichi BUG hotfixes that are not yet released in it's cycle.
```
Comment (yuanming-hu) [inline docs/external_libraries.rst:23]: Maybe this is not really the job that an extension library should do?

File: docs/taichicon.rst [hunk 6]
```diff
@@ -1,5 +1,5 @@
-TaichiCon
-=========
+Taichi Conferences
```
Comment (rexwangcc) [inline docs/taichicon.rst:1]: I'm not sure about this change. I thought `TaichiCon` was a proper noun, e.g. "DockerCon 2020", not "Docker Conference". Better to double-check with @yuanming-hu

File: docs/taichicon.rst [hunk 7]
```diff
@@ -1,5 +1,5 @@
...
+==================
```
[the remark this answers is not in the document]
Comment (archibate) [reply docs/taichicon.rst:2]: Users might confuse this to be a Taichi feature, e.g. TaichiSparse or TaichiDiff, let's be clear here.

File: docs/external_libraries.rst (hunk 5 above)
Comment (archibate) [reply docs/external_libraries.rst:23]: But that's what Taichi GLSL **have to do** before v0.6.11. Sometimes Taichi BUGs can limit Taichi GLSL's ability, e.g. docstring this time.

File: docs/external_libraries.rst [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+
+The solver is being developed using Taichi, therefore it is cross-platform and
+supports multithreaded CPUs and massively parallel GPUs.
+
+The short-term plan is:
+
+- To build a reusable multimaterial (water/elastic/snow/sand/mud) simulator
+- To integrate the simulator into Blender
+
+Click here for `Taichi Elements Documentation <https://taichi-elements.readthedocs.io>`_.
```
[the remark this answers is not in the document]
Comment (archibate) [reply docs/external_libraries.rst:43]: Wasn't it a helpful continum simulation infrastructure? Sorry, not sure about the work there...

File: docs/external_libraries.rst [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+
+Taichi GLSL
+-----------
+
+`Taichi GLSL <https://github.com/taichi-dev/taichi_glsl>`_ is an extension
+library of Taichi, aiming at providing useful helper functions including:
+
+1. Handy scalar functions like ``clamp``, ``smoothstep``, ``mix``, ``round``.
+2. GLSL-alike vector functions like ``normalize``, ``distance``, ``reflect``.
+3. Well-behaved random generators including ``randUnit2D``, ``randNDRange``.
```
Comment (Rullec) [inline docs/external_libraries.rst:15]: Would "random number generators" be better here? :D

**Comment by Rullec** [comment]
Nice job! LGTM and a few nits in the comments. :-)

File: docs/external_libraries.rst (hunk 9 above)
Comment (archibate) [reply docs/external_libraries.rst:15]: The rand unit in Taichi GLSL also supports random vectors, not sure if number will imply scalars..

File: docs/extension_libraries.rst [hunk 10]
```diff
@@ -0,0 +1,19 @@
...
+-----------
+
+`Taichi GLSL <https://github.com/taichi-dev/taichi_glsl>`_ is an extension
+library of Taichi, aiming at providing useful helper functions including:
+
+1. Handy scalar functions like ``clamp``, ``smoothstep``, ``mix``, ``round``.
+2. GLSL-alike vector functions like ``normalize``, ``distance``, ``reflect``.
+3. Well-behaved random generators including ``randUnit2D``, ``randNDRange``.
+4. Possible Taichi BUG hotfixes that are not yet released in it's cycle.
+5. Handy vector and matrix initializer: ``vec`` and ``mat``.
```
Comment (archibate) [inline docs/extension_libraries.rst:17]:
```suggestion
4. Handy vector and matrix initializer: ``vec`` and ``mat``.
```

File: docs/extension_libraries.rst [hunk 11]
```diff
@@ -0,0 +1,19 @@
...
+Taichi GLSL
+-----------
+
+`Taichi GLSL <https://github.com/taichi-dev/taichi_glsl>`_ is an extension
+library of Taichi, aiming at providing useful helper functions including:
+
+1. Handy scalar functions like ``clamp``, ``smoothstep``, ``mix``, ``round``.
+2. GLSL-alike vector functions like ``normalize``, ``distance``, ``reflect``.
+3. Well-behaved random generators including ``randUnit2D``, ``randNDRange``.
+4. Possible Taichi BUG hotfixes that are not yet released in it's cycle.
```
Comment (archibate) [inline docs/extension_libraries.rst:16]:
```suggestion
```

**Pull request merged**